### PR TITLE
fix(opencode-provider): correct z.ai coding plan model mapping

### PR DESCRIPTION
**Pull request by USerik**
The model mapping for 'z.ai coding plan' was incorrectly pointing to 'z-ai' instead of 'zai-coding-plan', which would cause model resolution failures when users selected the z.ai coding plan provider.

This fix ensures the correct model identifier is used for z.ai coding plan, aligning with the expected model naming convention.

## Summary by CodeRabbit

* **Bug Fixes**
  * Fixed provider identifier mapping so "z.ai coding plan" is correctly recognized as "zai-coding-plan", improving provider recognition.

* **Tests**
  * Added extensive unit tests covering provider name mapping, alias handling, ANSI/formatting robustness, authentication detection, and numerous edge cases to ensure parsing reliability.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: apps/server/src/providers/opencode-provider.ts [hunk 1]
```diff
@@ -1035,7 +1035,7 @@ export class OpencodeProvider extends CliProvider {
       'lm studio': 'lmstudio',
       lmstudio: 'lmstudio',
       opencode: 'opencode',
-      'z.ai coding plan': 'z-ai',
+      'z.ai coding plan': 'zai-coding-plan',
```
[the remark this answers is not in the document]
Comment (USerik) [reply apps/server/src/providers/opencode-provider.ts:1038]: Tests added

**Comment by stefandevo** [comment]
Recheck against current rc branch.
